forge/forge/op/loss.py

```diff
@@ -2,41 +2,41 @@
 
 # SPDX-License-Identifier: Apache-2.0
 from ..module import ForgeModule
-from .reduce import ReduceSum, ReduceAvg
-from .eltwise_unary import Exp, Log, Abs
-from .eltwise_binary import Subtract, Multiply
 from .constant import Constant
+from .eltwise_unary import Log, Abs
+from .eltwise_binary import Subtract, Multiply
+from .nn import Softmax
+from .reduce import ReduceSum, ReduceAvg
 
 
 class CrossEntropyLoss(ForgeModule):
     """
-    Cross-Entropy Loss
+    Cross-Entropy Loss (with mean reduction)
+
+    loss = reduce_avg(-1 * sum(labels * log(softmax(predictions)), dim=-1), dim=0)
     """
 
     def forward(self, predictions, labels):
-        exponential_predicted = Exp("exp", predictions)
-        reduced_exponential_predicted = ReduceSum("reduce_sum", exponential_predicted, dim=-1)
-        ln_reduced_exponential_predicted = Log("log", reduced_exponential_predicted)
-        output_minus_ln_reduced_exponential_predicted = Subtract(
-            "subtract", predictions, ln_reduced_exponential_predicted
-        )
+        assert predictions.ndim() == 2, f"Predictions must be a 2D tensor. Got {predictions.ndim()}."
+        assert (
+            predictions.shape == labels.shape
+        ), f"Shapes of predictions and labels must match. Got {predictions.shape} and {labels.shape}."
```

Comment:
Can we clarify the message?

More precisely, to explicitly say which shape is prediction, and which label